common/client/errors.go

```diff
@@ -0,0 +1,9 @@
+package client
+
+// Provides error classification to external components in a chain agnostic way
+// Only exposes the error types that could be set in the transaction error field
+type TxError interface {
```

Comment:
I would personally think of another way to return status.
Maybe we discuss this in the ChainWriter design?

Reply:
Based on the ChainWriter discussions, we wouldn't return the status using this interface but I think I would still need it internally to classify errors between the `Failed` and `Fatal` state in the common TXM code

Reply:
Although to avoid adding a new file, I don't think there's a reason for this to be in client. I can move this to the `common/txmgr/types/tx` file instead.